### PR TITLE
add ntia compliance report

### PR DESCRIPTION
**Pull request by viveksahu26**
closes: https://github.com/interlynk-io/sbomqs/issues/242

This PR will add NTIA minimum element compliance report. For now I have only updated README for NTIA minimum element compliance report.

### CODE REVIEW

**Comment by riteshnoronha** [comment]
@viveksahu26 this does not look accurate to me.

**Comment by viveksahu26** [comment]
No, it's not yet completed. It's in process. I have simply added NTIA compliance readme fow now, code part is still left.

**Comment by viveksahu26** [comment]
Hey @riteshnoronha , can you go through this NTIA minimum elements compliance report [readme](https://github.com/viveksahu26/sbomqs/blob/feat_ntia_compliance_report/Compliance.md) and let me know what changes to be made. The recommended data-fields are there in the NTIA minimum elements [report](https://www.ntia.doc.gov/files/ntia/publications/sbom_minimum_elements_report.pdf) - on page 15, so that's why I have added. We can also mark it as optional field.

Apart from that one thing I have noticed that the `cra_score.go` is 95% same as `oct_score.go` except some print statement. Similarly for `cra_report.go` and `oct_report.go`. And again for NTIA we have to repeat it and that would be `ntia_score.go` and `ntia_report.go`. Can we generalized these in a common package, which will contain both `score.go` and `report.go` functionality. WDT ??

**Comment by viveksahu26** [comment]
@riteshnoronha , any update here ?

**Comment by viveksahu26** [comment]
Hey @riteshnoronha, somewhat it's ready for review. Checkout this ntia compliance feature.  And if changes let me know.

File: Compliance.md [hunk 1]
```diff
@@ -66,3 +66,21 @@ The [OpenChain Telco](https://github.com/OpenChain-Project/Reference-Material/bl
...
 | SBOM Scope | 3.8 | `SBOM scope` | sbom scope | |
+
+## NTIA minimum elements: SBOM Requirements for NTIA
+
+The [NTIA](https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/TechGuidelines/TR03183/BSI-TR-03183-2.pdf) specifies mandatory properties for an SBOM. Below is how we have derived all the values.
+
+| NTIA minimum elements | Section ID | NTIA Fields | CycloneDX |SPDX(2.3)  | Notes |
+| :---   | :---  |:---   |   :---  |  :---  | :---   |
+| Automation Support | 1.1 | `Machine Readable Format` |  BomFormat & data forrmat | SPDXversion & data forrmat |  |
+| SBOM Data Fields | 2.1 |  `Author of the SBOM` | metadata->authors, metadata->supplier | creator  |   |
```
Comment (riteshnoronha) [inline Compliance.md:77]: For SPDX it should be creator->Person, creator->organization or creator->tool

File: Compliance.md [hunk 2]
```diff
@@ -66,3 +66,21 @@ The [OpenChain Telco](https://github.com/OpenChain-Project/Reference-Material/bl
...
+|  | 2.2 |  `Timestamp` |  metadata->timestamp  |  created  |   |
+|  | 2.3 |  `present` |           |           | all package elements |
```
Comment (riteshnoronha) [inline Compliance.md:79]: remove this

File: Compliance.md [hunk 3]
```diff
@@ -66,3 +66,21 @@ The [OpenChain Telco](https://github.com/OpenChain-Project/Reference-Material/bl
...
+| Package Data Fields | 2.4 |  `Package Name` |  component->name | package->name |  |
+|  | 2.5 |  `Dependency Relationship` | dependencies, composition | relationships  |  |
```
Comment (riteshnoronha) [inline Compliance.md:81]: Just dependencies here, composition is for depth.

File: Compliance.md [hunk 4]
```diff
@@ -66,3 +66,21 @@ The [OpenChain Telco](https://github.com/OpenChain-Project/Reference-Material/bl
...
+|  | 2.6 |  `Supplier Name`  |  component->supplier | packageSupplier, packageOriginator |  |
+|  | 2.7 |  `Version of Component` | component->version | package->version |  |
+|  | 2.8 |  `Other Uniq IDs` | component->cpe, component->purl | DocumentNamespace, SPDXID |  |
```
Comment (riteshnoronha) [inline Compliance.md:84]: For SPDX it should also be cpe/purl

File: Compliance.md [hunk 5]
```diff
@@ -66,3 +66,21 @@ The [OpenChain Telco](https://github.com/OpenChain-Project/Reference-Material/bl
 | Timing of SBOM delivery | 3.6 | `SBOM delivery time` | delivery time | |
 | Method of SBOM delivery | 3.7 | `SBOM delivery method` | delivery method | |
 | SBOM Scope | 3.8 | `SBOM scope` | sbom scope | |
+
+## NTIA minimum elements: SBOM Requirements for NTIA
```
Comment (riteshnoronha) [inline Compliance.md:70]: We should also denotes field which are mandatory vs optional with an *

**Review by riteshnoronha — CHANGES_REQUESTED** [review]

File: Compliance.md (hunk 5 above)
Comment (viveksahu26) [reply Compliance.md:70]: done

**Review by riteshnoronha — APPROVED** [review]

**Pull request merged**